src/controllers/trip.controller.js

```diff
@@ -0,0 +1,94 @@
+import * as ApplicationError from '../utils/errors/applicationsErrors';
+import { tripService, roleService } from '../services';
+import {notFoundResponse, successResponse,createdResponse} from '../utils/responseHandler';
+
+export const makeTripRequest = async (req, res) => {
+  const {departure,destination, dateOfTravel, dateOfReturn,travelReason, accomodationId
+  } = req.body;
+  const newTripRequest = {
+    departure,
+    destination,
+    dateOfTravel,
+    dateOfReturn,
+    travelReason,
+    accomodationId,
+    userId: req.user.id
+  };
+  const trip = await tripService.createTripRequest(newTripRequest);
+  if (trip) {
+    createdResponse(res, 'New trip request made successfully', trip);
+  } else {
+    ApplicationError.internalServerError(
+      `There was a problem making your trip request`,
+      res
+    );
+  }
+};
+
+export const getAllTripRequest = async (req, res) => {
+  if ((await roleService.getUserRole(req.user.id)) === 'Manager') {
+    const { trips, error } = await tripService.findTripByManagerId(req.user.id);
+    if (error) return ApplicationError.databaseError(error, res);
```

Comment:
do not assume any other failure will be a database error. sometimes it might be a logical error eg an operation on an invalid type